src/Cassandra.IntegrationTests/ExecutionProfiles/ExecutionProfileTests.cs

```diff
@@ -0,0 +1,199 @@
+//    You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+//
+
+using System.Linq;
+using System.Threading.Tasks;
+
+using Cassandra.IntegrationTests.TestClusterManagement.Simulacron;
+
+using NUnit.Framework;
+
+namespace Cassandra.IntegrationTests.ExecutionProfiles
+{
+    [TestFixture]
+    public class ExecutionProfileTests
+    {
+        private SimulacronCluster _simulacron;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _simulacron = SimulacronCluster.CreateNew(3);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            _simulacron.Dispose();
+        }
+
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public void Should_UseDerivedProfileConsistency_When_DerivedProfileIsProvided(bool async)
+        {
+            var writeProfile =
+                Builder.ExecutionProfileBuilder()
+                                .WithLoadBalancingPolicy(new RoundRobinPolicy())
+                                .WithConsistencyLevel(ConsistencyLevel.All)
+                                .Build();
+            var readProfile =
+                Builder.ExecutionProfileBuilder()
+                                .WithConsistencyLevel(ConsistencyLevel.Two)
+                                .Build();
+            var cluster =
+                Cluster.Builder()
+                       .AddContactPoint(_simulacron.InitialContactPoint)
+                       .WithQueryOptions(new QueryOptions().SetConsistencyLevel(ConsistencyLevel.One))
+                       .WithExecutionProfiles(opts => opts
+                           .WithProfile("write", writeProfile)
+                           .WithDerivedProfile("read", "write", readProfile))
+                       .Build();
+            var session = cluster.Connect();
+            var primeQuery = new
+            {
+                when = new { query = "SELECT * from test.test", consistency_level = new[] { "ALL" } },
+                then = new
+                {
+                    result = "unavailable",
+                    consistency_level = (int)ConsistencyLevel.All,
+                    required = 3,
+                    alive = 2,
+                    delay_in_ms = 0,
+                    message = "unavailable",
+                    ignore_on_prepare = false
+                }
+            };
+            _simulacron.Prime(primeQuery);
+
+            var exception = async
+                ? Assert.ThrowsAsync<UnavailableException>(() => session.ExecuteAsync(new SimpleStatement("SELECT * from test.test"), "write"))
```

Comment:
I'm not sure I understand, shouldn't `"write"` exec profile use consistency level one and not fail?

Reply:
Ah I primed the request for the `"write"` profile and I'm using the `"write"` profile in the test but the derived profile is `"read"`. Changed the rtest to prime for consistency `"TWO"` and use `"read"` in the test.